modules/custom/openy_activity_finder/js/programs_search.js

```diff
@@ -394,45 +491,71 @@
           component.table = data.table;
           component.count = data.count;
           component.pages[component.current_page + 1] = data.pager;
+          component.pager_info = data.pager_info;
+          component.no_results = data.count > 0 ? '0' : '1';
+          component.sort = data.sort;
+
+          router.push({ query: {
+            locations: cleanLocations.join(','),
+            categories: cleanCategories.join(','),
+            ages: cleanAges.join(','),
+            days: cleanDays.join(','),
+            keywords: component.keywords,
+            page: component.page,
+            no_results: component.no_results,
+            sort: component.sort,
+          }});
+        }).done(function() {
+          // We need to wait in order to affect the DOM after the tiles have been injected.
           component.loading = false;
         });
-
-        router.push({ query: {
-          locations: cleanLocations.join(','),
-          categories: cleanCategories.join(','),
-          ages: cleanAges.join(','),
-          days: cleanDays.join(','),
-          keywords: this.keywords,
-          page: this.page
-        }});
+      },
+      searchAlternativeResults: function(type) {
+        this.alternativeCriteria = type;
+        this.runAjaxRequest();
       },
       populatePopupLocation: function(index) {
         this.locationPopup = this.table[index].location_info;
       },
+      convertAdressField: function(adress) {
+        adress = adress.split(',').map(function(item) {return item.trim()});
+        let adress_out = adress[0] ? adress[0] : '';
+        adress_out += '<br />';
+        adress_out += adress[1] ? adress[1] : '';
+        adress_out += ', ';
+        adress_out += adress[2] ? adress[2] : '';
+        adress_out += ' ';
+        adress_out += adress[3] ? adress[3] : '';
+        return adress_out;
```

Comment:
could you please remove possible duplicates of `,` ( comma )?
I see not all fields are filled in CRM, so we should be more clever with forming a string

Reply:
fixed